src/components/attendance/attendance.module.ts

```diff
@@ -0,0 +1,27 @@
+import { MiddlewareConsumer, Module } from '@nestjs/common';
+import { AttendanceController } from './attendance.controller';
+import { AttendanceService } from './attendance.service';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { AttendanceEntity } from './entity/attendance.entity';
+import { StudentModule } from '../student/student.module';
+import { StudentEntity } from '../student/entity/student.entity';
+import { AttendanceRepository } from './attendance.repository';
+import { AuthenticationMiddleware } from '../../middlewear/auth.middlewar';
+import { UserRepository } from '../user/user.repository';
+import { UserModule } from '../user/user.module';
+
+@Module({
+  imports: [
+    TypeOrmModule.forFeature([AttendanceEntity, StudentEntity]),
+    StudentModule,
+    UserModule,
+  ],
+  controllers: [AttendanceController],
+  providers: [AttendanceService, AttendanceRepository, UserRepository],
+  exports: [TypeOrmModule, AttendanceRepository],
```

Comment:
### It is required else it will throw an error
**using for feature we register the entity for that particular module now that entity to be used by another module for db operations we need to put typeorm module in  exports array so that entity registered in it can be used by another modules !**